datalad_next/create_sibling_webdav.py

```diff
@@ -422,41 +429,107 @@ def _create_sibling_webdav(
+        else:
+            yield _create_storage_sibling(
+                    ds,
+                    url,
+                    storage_name,
+                    credential,
+                    export=export_storage,
+                    reconfigure=False,
+                )
+
     if 'only' not in storage_sibling:
-        yield from _create_git_sibling(
-            ds,
-            url,
-            name,
-            existing,
-            credential_name,
-            credential,
-            export=export_storage,
-            dependency=None if storage_sibling == 'no' else storage_name,
-        )
+        if name in existing_siblings:
+            yield from maybe_skip_sibling(credential, credential_name, ds,
+                                          existing, export_storage, name,
+                                          storage_name, storage_sibling,
+                                          url, 'git')
+        else:
+            yield _create_git_sibling(
+                    ds,
+                    url,
+                    name,
+                    credential_name,
+                    credential,
+                    export=export_storage,
+                    reconfigure=False,
+                    publish_depends=storage_name if storage_sibling != 'no'
+                    else None
+            )
+
+
+def maybe_skip_sibling(credential, credential_name, ds, existing,
+                       export_storage, name, storage_name, storage_sibling,
+                       url, sibling_type):
+    """
+
+    Parameters:
+    -----------
+    sibling_type: str
+      'git' or 'storage'
+    """
+    if existing == 'skip':
+        yield get_status_dict(
+            ds=ds,
+            status='notneeded',
+            message=f"Skipped on existing sibling "
+                    f"{name if sibling_type == 'git' else storage_name}")
+    elif existing == 'reconfigure':
+        if sibling_type == 'git':
+            yield _create_git_sibling(
+                ds,
+                url,
+                name,
+                credential_name,
+                credential,
+                export=export_storage,
+                reconfigure=True,
+                publish_depends=storage_name
+                                if storage_sibling != 'no' else None
+            )
+        elif sibling_type == 'storage':
+            yield _create_storage_sibling(
+                ds,
+                url,
+                storage_name,
+                credential,
+                export=export_storage,
+                reconfigure=True,
+            )
+    else:
+        # Shouldn't happen, since 'error' was treated upfront
+        raise ValueError(f"Unexpected value of 'existing': {existing}")
 
 
 def _create_git_sibling(
-        ds, url, name, existing, credential_name, credential, export,
-        dependency=None):
+        ds, url, name, credential_name, credential, export, reconfigure,
+        publish_depends=None):
```

Comment:
I passed `existing`, because I was thinking about future patches, and what information would be valuable in this function. Now it is limited to handle the reconfigure case, and no, e.g., `replace` that may come in the not too distant future.

Reply:
Done